app.py

```diff
@@ -241,6 +230,40 @@ def _get_contacts_data():
     return _contacts_data
 
 
+def _get_authorized():
+    """
+    Determine if the client is authorized
+
+    returns: True if authorized, False otherwise
+    """
+    if 'GRST_CRED_AURI_0' in request.environ:
```

Comment:
We should iterate through all the `GRST_CRED_AURI_` headers instead... I'm not sure we're guaranteed that the DN itself is in `0`.